http_types/utils.py

```diff
@@ -237,12 +261,63 @@ def validate(reqres: HttpExchange) -> None:
         """
         check_type("request-response", reqres, HttpExchange)
 
+
+class HttpExchangeReader:
+
+    def __init__(self):
+        raise Exception("Do not instantiate")
+
     @staticmethod
-    def from_jsonl(input_file: IO) -> Generator[HttpExchange, None, None]:
-        """Read Http exchanges line by line from a file-like object.
+    def from_json(input_json: Union[str, bytes]) -> HttpExchange:
+        """Read a single HTTP exchange from a JSON string.
+
+        Arguments:
+            input_json: {str} -- The input JSON to parse.
+        """
+        return HttpExchangeBuilder.from_dict(json.loads(input_json))
+
+    @staticmethod
+    def from_jsonl(input_file: IO[str]) -> Generator[HttpExchange, None, None]:
+        """Read HTTP exchanges line by line from a file-like object.
 
         Arguments:
             input_file: {IO} -- The input to read from.
         """
         for line in input_file:
-            yield HttpExchangeBuilder.from_dict(json.loads(line))
+            yield HttpExchangeReader.from_json(line)
+
+
+def json_serial(obj):
+    """JSON serializer for objects not serializable by default json code"""
+
+    if isinstance(obj, datetime):
+        return obj.isoformat()
+    raise TypeError ("Type %s not serializable" % type(obj))
+
+
+class HttpExchangeWriter:
+
+    def __init__(self, output: IO[str]):
+        """Create a writer of HTTP exchanges.
+
+        Each exchange will be written as a JSON object in the HTTP types
+        format on a single line.
+
+        Arguments:
+            output: {IO} -- The output to write to
+        """
+        self.output = output
+        self.first = True
+
+    def write(self, exchange: HttpExchange):
+        """Write a single HTTP exchange line to the output.
+
+        Arguments:
+            exchange: {HttpExchange} -- The exchange to write.
+        """
+        if self.first:
+            self.first = False
+        else:
+            self.output.write("\n")
```

Comment:
Could one add a line-break after every written line instead of adding them when writing a new object? Just asking because that's what I'm always doing :)

Reply:
That's probably cleaner :), I updated to it! And http://jsonlines.org/ is fine with that:

> This means '\r\n' is also supported because trailing white space is ignored when parsing JSON values.
> The last character in the file may be a line separator, and it will be treated the same as if there was no line separator present.